### PR TITLE
Jenkins improvements

### PR DESCRIPTION
**Pull request by tjhei**
- report why CI failed (copied from OSX)
- if not "ready to test" report pending not failure (no red X)
- abort older builds when starting up (copied from OSX)
- formatting

Not sure if we should wait until the hack is over, to make sure we don't break things.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,186 +1,247 @@
...
+            githubNotify context: 'Jenkins', description: 'tests failed',  status: 'FAILURE'
+          }
         }
       }
     }
   }
-  post {
-    cleanup {
-      cleanWs()
-    }
```
Comment (gassmoeller) [inline Jenkinsfile:184]: No need for file cleanup anymore?

**Review by gassmoeller — COMMENTED** [review]
Just one question. I would be fine with merging it right away, we always have the github actions tester, which tests almost the same.

**Comment by gassmoeller** [comment]
Looks like it is not working though.

**Comment by tjhei** [comment]
seems to be working now.

**Review by gassmoeller — APPROVED** [review]

**Pull request merged**